src/Analyzer.Core/TemplateAnalyzer.cs

```diff
@@ -114,42 +166,93 @@ public IEnumerable<IEvaluation> AnalyzeTemplate(string template, string template
 
             var templateContext = new TemplateContext
             {
-                OriginalTemplate = JObject.Parse(template),
+                OriginalTemplate = JObject.Parse(populatedTemplate),
                 ExpandedTemplate = templatejObject,
-                IsMainTemplate = true,
+                IsMainTemplate = parentContext.ParentContext == null,
                 ResourceMappings = armTemplateProcessor.ResourceMappings,
                 TemplateIdentifier = templateFilePath,
-                IsBicep = isBicep,
-                SourceMap = sourceMap
+                IsBicep = parentContext.IsBicep,
+                SourceMap = parentContext.SourceMap,
+                PathPrefix = pathPrefix,
+                ParentContext = parentContext
             };
 
             try
             {
                 IEnumerable<IEvaluation> evaluations = this.jsonRuleEngine.AnalyzeTemplate(templateContext);
                 evaluations = evaluations.Concat(this.powerShellRuleEngine.AnalyzeTemplate(templateContext));
 
-                // For each rule we don't want to report the same line more than once
-                // This is a temporal fix
-                var evalsToValidate = new List<IEvaluation>();
-                var evalsToNotValidate = new List<IEvaluation>();
-                foreach (var eval in evaluations)
+                // Recursively handle nested templates 
+                var jsonTemplate = JObject.Parse(populatedTemplate);
+                var processedTemplateResources = templatejObject.InsensitiveToken("resources");
+
+
+                for (int i = 0; i < processedTemplateResources.Count(); i++)
                 {
-                    if (!eval.Passed && eval.Result != null)
-                    {
-                        evalsToValidate.Add(eval);
-                    }
-                    else
+                    var currentProcessedResource = processedTemplateResources[i];
+
+                    if (currentProcessedResource.InsensitiveToken("type").ToString().Equals("Microsoft.Resources/deployments", StringComparison.OrdinalIgnoreCase))
```

Comment:
Use `?` in case "type" returns a null object.  `InsensitiveToken("type")?.ToString()`.  Since this will result in a `bool?` (nullable Boolean), it won't work by itself in the `if` statement, so you'll need to use the "null-coalescing" operator (`??`) to choose a Boolean value in case it's null. i.e. if "type" doesn't exist, don't go into the `if` statement.
```cs
                    if (currentProcessedResource.InsensitiveToken("type")?.ToString().Equals("Microsoft.Resources/deployments", StringComparison.OrdinalIgnoreCase) ?? false)
```